src/userbase-js/types/index.d.ts

```diff
@@ -0,0 +1,70 @@
+/* auth */
+
+export interface Session {
+  sessionId: string
+  creationDate: string
+}
+
+export interface UserProfile {
+  [key: string]: string
+}
+
+export interface UserResult {
+  username: string
+  key: string
+  email?: string
+  profile?: UserProfile
+}
+
+export type KeyNotFoundHandler = (username: string, deviceId: string) => void
+
+export type ShowKeyHandler = (seedString: string, rememberMe: boolean, backUpKey: boolean) => void | Promise<void>
+
+export function init(options: { appId: string, endpoint?: string, keyNotFoundHandler?: KeyNotFoundHandler }): Promise<Session>
+
+export function signUp(username: string, password: string, email?: string, profile?: UserProfile, showKeyHandler?: ShowKeyHandler, rememberMe?: boolean, backUpKey?: boolean): Promise<UserResult>
+
+export function signIn(username: string, password: string, rememberMe?: boolean): Promise<UserResult>
+
+export function signOut(): Promise<void>
+
+export function forgotPassword(username: string): Promise<void>
+
+export function updateUser(user: { username?: string, password?: string, email?: string | null, profile?: UserProfile | null }): Promise<void>
+
+export function deleteUser(): Promise<void>
+
+export function importKey(keyString: string): Promise<void>
+
+export function getLastUsedUsername(): string | undefined
+
+/* db */
+
+export type DatabaseOperation = InsertOperation | UpdateOperation | DeleteOperation
+
+export interface InsertOperation {
+  command: 'Insert'
+  id?: string
+  item: any
+}
+
+export interface UpdateOperation {
+  command: 'Update'
+  id: string
+  item: any
+}
+
+export interface DeleteOperation {
+  command: 'Delete'
+  id: string
+}
+
+export function openDatabase(dbName: string, changeHandler: (items: any[]) => void): Promise<void>
```

Comment:
@j-berman what is the shape of items that get passed to the changeHandler callback in openDatabase?

Something like [ { itemId: string, item: any } ]? This is what it seems like going off the ugly todo tutorial code but having a hard time being sure from the source code.

Reply:
`[ { itemId: string, item: any } ]` is accurate

[source 1](https://github.com/encrypted-dev/userbase/blob/172ce59f6c69b7d5d3ae17fde04d3865d29c259c/src/userbase-js/db.js#L327)
[source 2 that shows item can be any](https://github.com/encrypted-dev/userbase/blob/172ce59f6c69b7d5d3ae17fde04d3865d29c259c/src/userbase-js/db.js#L152-L154)

Reply:
Updated